Assets/_Game/Scripts/UI/ActivationPopupUI.cs

```diff
@@ -0,0 +1,86 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+public class ActivationPopupUI : MonoBehaviour
+{
+    [SerializeField]
+    private Animator anim;
+    [SerializeField]
+    private PlayerInput playerInput;
+
+    private void OnEnable()
+    {
+        if(playerInput != null)
+        {
+            playerInput.onControlsChanged += PlayerInput_onControlsChanged;
+
+            HandleInput(playerInput);
+        }
+    }
+
+    private void OnDisable()
+    {
+        if (playerInput != null)
+        {
+            playerInput.onControlsChanged -= PlayerInput_onControlsChanged;
+        }
+    }
+
+    private void Awake()
+    {
+        anim = GetComponent<Animator>();
+        playerInput = FindObjectOfType<PlayerInput>();
+    }
+
+    private void PlayerInput_onControlsChanged(PlayerInput input)
+    {
+        HandleInput(input);
+    }
+
+    public void HandleInput(PlayerInput input)
+    {
+        if (input.currentControlScheme == "Keyboard&Mouse") { ChangeUIPrompt("KEYBOARD"); }
+        if (input.currentControlScheme == "Gamepad")
+        {
+            if (Gamepad.current.device.name.Contains("XInput")) { ChangeUIPrompt("XINPUT"); }
+            else if (Gamepad.current.device.name.Contains("DualShock")) { ChangeUIPrompt("DUALSHOCK"); }
+            else if (Gamepad.current.device.name.Contains("Switch")) { ChangeUIPrompt("SWITCH"); }
+        }
+    }
+
+    public void ChangeUIPrompt(string deviceName)
+    {
+
+        switch (deviceName.ToUpper())
+        {
+
+            case "XINPUT":
+                anim.SetBool("isKeyboard", false);
+                anim.SetBool("isGamepadXbox", true);
+                anim.SetBool("isGamepadPlaystation", false);
+                anim.SetBool("isGamepadSwitch", false);
+                break;
+            case "SWITCH":
+                anim.SetBool("isKeyboard", false);
+                anim.SetBool("isGamepadXbox", false);
+                anim.SetBool("isGamepadPlaystation", false);
+                anim.SetBool("isGamepadSwitch", true);
+                break;
+            case "DUALSHOCK":
+                anim.SetBool("isKeyboard", false);
+                anim.SetBool("isGamepadXbox", false);
+                anim.SetBool("isGamepadPlaystation", true);
+                anim.SetBool("isGamepadSwitch", false);
+                break;
+            case "KEYBOARD":
+            default:
+                anim.SetBool("isKeyboard", true);
+                anim.SetBool("isGamepadXbox", false);
+                anim.SetBool("isGamepadPlaystation", false);
+                anim.SetBool("isGamepadSwitch", false);
```

Comment:
If each case is only handling setting one bool true and the others false, you could do a code cleanup to just have the 4 settings and pass in if the case equality as the second parameter. 
IE 
`anim.SetBool("isKeyboard", deviceName.ToUpper().Equals("KEYBOARD");`

but I'd cache the deviceName.ToUpper  first

Reply:
changed to enum but removed the switch statement.